source/assets/public/index.js

```diff
@@ -1,33 +1,22 @@
-/**
- * JavaScript goes here
- */
-$(document).ready(function () {
-  mobileNav();
-});
+var body = document.body;
+var mobileNavContainer = document.querySelector('.nav--mobile');
+var mobileNavTrigger = document.querySelector('.nav--mobile__show');
 
-function hide ($element) {
-  $($element).hide(200);
-  $($element).removeClass('active');
+function bodyClickHandler(event) {
+  var trigger = event.target;
+  if (trigger == mobileNavTrigger) {return;}
+  removeClass();
 }
-
-function mobileNav () {
-  var $mobileNav = $('.nav--mobile');
-
-  $('.nav--mobile__show').click(function () {
-    if ( $mobileNav.hasClass('active') ) {
-      hide($mobileNav);
-      return;
-    }
-
-    $mobileNav.addClass('active');
-    $mobileNav.show(200);
-  });
-
-  $('body').click(function (event) {
-    if (!$mobileNav.hasClass('active')) { return; }
-    if ($(event.target).is('.nav--mobile__show') || $(event.target).parent().is('.nav--mobile__show')) { return; }
-
-    $($mobileNav).removeClass('active');
-    hide($mobileNav);
-  });
+function addClass () {
+  mobileNavContainer.classList.add('active');
+  body.addEventListener('click', bodyClickHandler);
+}
+function removeClass () {
```

Comment:
I think it's worth either passing the class name into this function or giving it a more specific name